mesonbuild/mesonmain.py

```diff
@@ -112,8 +112,9 @@ def add_help_arguments(self, parser):
         parser.add_argument('command', nargs='?')
 
     def run_help_command(self, options):
-        if options.command:
-            self.commands[options.command].print_help()
+        command = self.commands.get(options.command)
+        if command:
+            command.print_help()
```

Comment:
This means that if someone runs `meson help gargle` meson will simply return the help for `meson help` because it treats it identical to no arguments.

I'm not sure that's optimal, e.g. if you pass an invalid option to a different subcommand it returns an error.

Maybe instead we could build a new set of command line arguments, which passes the command (if anything) followed by `--help` into `self.parser.parse_args()`?

Reply:
Yes ofcourse, we can add choices option to display available commands.
I'll refactor the code for the same. Thanks.